src/data_api/registrationToken.se

```diff
@@ -0,0 +1,113 @@
+
+def init():
+    self.controller = 0x0
+    # TODO: make this dynamically adjust based on the rate of bond forfeitures in previous preporting window(s)
+    self.bondAmount = 10**18
+
+def any():
+    refund()
+
+def register(branch, reportingWindow):
+    # do not allow for registration for reporting in the current window
+    if (reportingWindow <= self.controller.lookup('reporting').getCurrentReportingWindow()):
+        ~invalid()
+    self.controller.lookup('reputationToken').transferFrom(branch, msg.sender, self, bondAmount)
+    startingTokens = self.tokens[branch][reportingWindow][msg.sender]
+    self.tokens[branch][reportingWindow][msg.sender] = safeAdd(startingRegistrationTokens, 1)
+
+def migrateRegistrationToken(parentBranch, childBranch, reportingWindow, tokensToMigrate):
+    branchesContract = self.controller.lookup('branches')
+    # validate that the source branch is actually forked/forking
+    parentForkingEvent = branchesContract.getForkingEvent(parentBranch)
+    if (parentForkingEvent == 0):
+        ~invalid()
+    # validate that destinationBranch is a child of sourceBranch
+    sourceBranchHashHighBits = sha3(sourceBranch) & 0xffffffffffffffff000000000000000000000000000000000000000000000000
+    destinationBranchParentHashHighBits = destinationBranch & 0xffffffffffffffff000000000000000000000000000000000000000000000000
+    if (sourceBranchHashHighBits != destinationBranchParentHashHighBits):
+        ~invalid()
+    # do the transfer
+    startingTokensOnParent = self.tokens[parentBranch][reportingWindow][msg.sender]
+    startingTokensOnChild = self.tokens[childBranch][reportingWindow][msg.sender]
+    startingSupplyOnParent = self.tokenSupply[parentBranch][reportingWindow]
+    startingSupplyOnChild = self.tokenSupply[childBranch][reportingWindow]
+    self.tokens[parentBranch][reportingWindow][msg.sender] = safeSub(startingTokensOnParent, tokensToMigrate)
+    self.tokens[childBranch][reportingWindow][msg.sender] = safeAdd(startingTokensOnChild, tokensToMigrate)
+    self.tokenSupply[parentBranch][reportingWindow] = safeSub(startingSupplyOnParent, tokensToMigrate)
+    self.tokenSupply[childBranch][reportingWindow] = safeAdd(startingSupplyOnChild, tokensToMigrate)
+    return(1)
+
+def redeem(branch, reportingWindow):
+    startingTokens = self.tokens[branch][reportingWindow][msg.sender]
+    startingSupply = self.tokenSupply[branch][reportingWindow]
+    reportingContract = self.controller.lookup('reporting')
+    reportingContract.assertDoneReporting(branch, reportingWindow, msg.sender)
+    self.tokens[branch][reportingWindow][msg.sender] = safeSub(startingTokens, 1)
+    self.tokenSupply[branch][reportingWindow] = safeSub(startingSupply, 1)
+    reportingContract.clearUserReports(branch, reportingWindow, msg.sender)
+    self.controller.lookup('reputation').transfer(branch, msg.sender, self.bondAmount)
```

Comment:
Just a reminder if bond is dynamic then it'll need to be something like self.bondAmount[reportingWindow]